### PR TITLE
Chore/update legal hold to api clients

### PR DESCRIPTION
**Pull request by tora-kozic**
### Description of Change ###

Add the legal hold endpoints which are supported by API Client authentication.  

To retain backwards compatibility, the existing legal hold client should be available when the SDK is initialized with any other authentication form.  If initialized with API Client auth, then the new updated legal hold client should be attached and these new endpoints should be available. 

Because this is a separate client, there are also new sets of unit and integration tests, as well as some small clarifying changes to variable names and error messages.

Breaking changes from the past client can be found in the CHANGELOG.

Because API Clients is not being released in py42 quite yet, this will target the `feature/api-clients` branch for merging.

### Testing Procedure ###
To initialize the sdk with API Client auth, use `py42.sdk.from_api_client(host, client, secret)`

the `client` and `secret` values can be created in the [console](https://console.core-int.cloud.code42.com/app/#/admin/api-clients)

### PR Checklist ###
Did you remember to do the below?

- [x] Add unit tests to verify this change
- [x] Add an entry to CHANGELOG.md describing this change
- [x] Add docstrings for any new public parameters / methods / classes

### CODE REVIEW

File: src/py42/services/legalholdapiclient.py
Comment (tora-kozic) [inline src/py42/services/legalholdapiclient.py:24]: It may be nicer to stick this into the `services/legalhold.py` file to prevent some redundancies and not have these wordy file names - but this follows our convention of a file for each service.

File: src/py42/sdk/__init__.py
Comment (tora-kozic) [inline src/py42/sdk/__init__.py:368]: Setting a flag which specifies how the sdk was initialized with the simplest solution I came up with for switching between which client is used - but I'm open to other solutions if there was something else in mind.

**Comment by tora-kozic** [comment]
*Integration tests will pass once mock-server changes are approved and merged. I think CLA Assistant is failing due to the target merge branch not being `main`*

**Comment by peterbriggs42** [comment]
Looks like the integration tests are failing because https://github.com/code42/code42-mock-servers/pull/20 hasn't merged yet. Is that PR ready to go? I think we should probably have that merge before this one.

File: src/py42/services/legalholdapiclient.py
Comment (ryan-haley-code42) [inline src/py42/services/legalholdapiclient.py:34]: is v27 correct here? I know we don't follow the typical versioning, but I feel like I should still ask.

**Review by ryan-haley-code42 — APPROVED** [review]
Just the one v27 comment, but I assume it's correct. I'll approve here so this can be merged when the ends up being correct.

File: src/py42/services/legalholdapiclient.py
Comment (tora-kozic) [reply src/py42/services/legalholdapiclient.py:34]: Yeah, this isn't our versioning scheme, these endpoints are provided by the microservice (or in this case core) and don't go through the gateway.

File: src/py42/services/legalholdapiclient.py
Comment (alanag13) [reply src/py42/services/legalholdapiclient.py:24]: I generally err on the side of less scrolling

File: src/py42/sdk/__init__.py
Comment (alanag13) [reply src/py42/sdk/__init__.py:368]: The general approach is fine but I think I would just use a boolean instead of defining a class/constant for this (I don't anticipate that we will be adding additional authentication methods that will require new API implementations any time soon).

**Pull request merged**
